src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -293,131 +203,71 @@ export default class DefaultDeviceController
-          this.transform.nodes.end.disconnect();
-          this.transform.nodes = undefined;
-        }
-        this.transform = undefined;
-      }
+    try {
+      if (isAudioTransformDevice(device)) {
+        // N.B., do not JSON.stringify here — for some kinds of devices this
+        // will cause a cyclic object reference error.
+        this.logger.info(`Choosing transform input device ${device}`);
 
-      /* istanbul ignore else */
-      if (this.audioInputSourceNode) {
-        this.audioInputSourceNode.disconnect();
-        this.audioInputSourceNode = undefined;
+        await this.chooseAudioTransformInputDevice(device);
+      } else {
+        this.logger.info(`Choosing intrinsic audio input device ${device}`);
+        this.removeTransform();
+        await this.chooseInputIntrinsicDevice('audio', device);
       }
-
-      /* istanbul ignore else */
-      if (this.audioInputDestinationNode) {
-        this.audioInputDestinationNode.disconnect();
-        this.audioInputDestinationNode = undefined;
+      this.trace('startAudioInputDevice', device, `success`);
+      // For web audio, the audio destination stream stays the same so audio input did not change
+      if (this.useWebAudio) {
+        this.attachAudioInputStreamToAudioContext(this.activeDevices['audio'].stream);
+        this.pushAudioMeetingStateForPermissions(this.getMediaStreamDestinationNode().stream);
+        return this.getMediaStreamDestinationNode().stream;
+      } else {
+        this.publishAudioInputDidChangeEvent(this.activeDevices['audio'].stream);
+        return this.activeDevices['audio'].stream;
       }
-
-      DefaultDeviceController.closeAudioContext();
+    } catch (error) {
+      throw error;
+    } finally {
+      this.getAudioInputInProgress = false;
     }
+  }
 
-    if (isAudioTransformDevice(device)) {
-      // N.B., do not JSON.stringify here — for some kinds of devices this
-      // will cause a cyclic object reference error.
-      this.logger.info(`Choosing transform input device ${device}`);
-
-      await this.chooseAudioTransformInputDevice(device);
-    } else {
-      this.logger.info(`Choosing intrinsic audio input device ${device}`);
-      this.removeTransform();
-      await this.chooseInputIntrinsicDevice('audio', device, false);
-      this.trace('chooseAudioInputDevice', device, `success`);
+  async stopAudioInput(): Promise<void> {
+    if (this.getAudioInputInProgress) {
+      throw new Error('An audio input is starting');
     }
-
-    // Only recreate if there's a peer connection, otherwise `restartLocalAudio` will throw.
-    // This hack is off by default, so tests don't cover it. We can remove this skip soon.
-    /* istanbul ignore next */
-    if (recreateAudioContext && this.boundAudioVideoController?.rtcPeerConnection) {
-      this.boundAudioVideoController.restartLocalAudio(() => {
-        this.logger.info('Local audio restarted.');
-      });
+    try {
+      if (this.useWebAudio) {
```

Comment:
So, will there be cases where Web audio is used but no audio transform device is been used? Earlier, we would explicitly check for `isAudioTransformDevice(device)`.

May be using an explicit check for transform devices is better over web audio check? Also, say I turned Web audio ON, but never selected or enabled VF, does this stop correctly then?

Reply:
So WebAudio can be used without transform device (basically audio transform needs webAudio but not vice versa).